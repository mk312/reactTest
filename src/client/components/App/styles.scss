.wrapper {
  border: 2px solid #C71585;
}