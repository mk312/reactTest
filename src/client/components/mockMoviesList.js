const moviesList = {
  data: [
    {
      id: 1,
      title: 'Kill Bill-1',
      tagline: 'Oscar winning',
      vote_average: 5,
      vote_count: 0,
      release_date: '2000',
      poster_path: 'media/51fY8NCyXpL._SX342_.jpg',
      overview: ' American martial arts film written and directed by Quentin Tarantino. '
      + 'It stars Uma Thurman as the Bride, who swears revenge on a team of assassins (Lucy Liu, Michael Madsen, '
      + 'Daryl Hannah, and Vivica A. Fox) and their leader, Bill (David Carradine), after they try to kill her and her '
      + 'unborn child. Her journey takes her to Tokyo, where she battles the yakuza.',
      budget: 0,
      revenue: 0,
      runtime: 154,
      genres: [
        'Drama',
      ],
    },
    {
      id: 2,
      title: 'Kill Bill-2',
      tagline: 'string',
      vote_average: 0,
      vote_count: 0,
      release_date: '2002',
      poster_path: 'media/51fY8NCyXpL._SX342_.jpg',
      overview: 'string',
      budget: 0,
      revenue: 0,
      runtime: 0,
      genres: [
        'Drama',
        'Thriller',
      ],
    },
  ],
};

export default moviesList;
