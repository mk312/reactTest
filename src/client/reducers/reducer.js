import { initialState } from '../initialState';
import {
    UPDATE_MOVIES_ARR,
    INCREASE_MOVIES_ARR,
    UPDATE_SEARCH_WORD,
    UPDATE_FILTER,
    UPDATE_SORTING,
    UPDATE_MOVIES_PAGE,
    SHOW_MOVIE
} from '../actions/actions'

export function reducer (state = initialState, action ) {
    // console.log('action->', action, ' state->', state);

    switch (action.type) {
        case UPDATE_MOVIES_ARR:
            return {
                ...state,
                moviesList: action.moviesArr,
            };

        case INCREASE_MOVIES_ARR:
            return {
                ...state,
                moviesList: state.moviesList.concat(action.moviesArr)
            };

        case UPDATE_SEARCH_WORD:
            return {
                ...state,
                searchParams : {
                    ...state.searchParams,
                    searchWord: action.searchWord
                },
            };

        case UPDATE_FILTER:
            return {
                ...state,
                searchParams : {
                    ...state.searchParams,
                    filterBy: action.filterBy
                },
            };

        case UPDATE_SORTING:
            return {
                ...state,
                searchParams : {
                    ...state.searchParams,
                    sortBy: action.sortBy
                },
            };
        case UPDATE_MOVIES_PAGE:
            return {
                ...state,
                searchParams: {
                    ...state.searchParams,
                    offset: action.offset
                },
            };
        case SHOW_MOVIE:
            return {
                ...state,
                chosenMovie: action.chosenMovie,
            };
        default:
            return state;
    }
}
