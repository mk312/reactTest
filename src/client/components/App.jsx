import React, { Component } from 'react';
import Article from './Article.jsx';
import Summary from './Summary.jsx';
import Title from './Title.jsx';
import PostScriptum from './PostScriptum.jsx';

import styles from './scss/styles.scss';

export default class App extends Component {
    constructor(props) {
        super(props);
        this.username = 'World'
    }

    render() {
        return (
            <div>
                <em>details: {this.props.info}</em>
                <Title username={this.username}/>
                <Article username={this.username} text='Where have you been?'/>
                <Article username={this.username} text='We were missing you...'/>
                <Summary username={this.username}/>
                <PostScriptum username={this.username}/>
            </div>
        );
    }
}