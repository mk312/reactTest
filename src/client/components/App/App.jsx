import styles from './styles.scss';

import React, { Component } from 'react';
import Article from '../Article/Article.jsx';
import Summary from '../Summary/Summary.jsx';
import Title from '../Title/Title.jsx';
import PostScriptum from '../PostScriptum/PostScriptum.jsx';

export default class App extends Component {
    constructor(props) {
        super(props);
        this.username = 'World'
    }

    render() {
        return (
            <div className={styles.wrapper}>
                <em>details: {this.props.info}</em>
                <Title username={this.username}/>
                <Article username={this.username} text='Where have you been?'/>
                <Article username={this.username} text='We were missing you...'/>
                <Summary username={this.username}/>
                <PostScriptum username={this.username}/>
            </div>
        );
    }
}