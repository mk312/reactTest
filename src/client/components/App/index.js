import React, { Component } from 'react';
import '@babel/polyfill';
import { moviesList } from './../mockMoviesList';
import { connect } from 'react-redux';
import {showMovie} from '../../actions/actions';

import ErrorBoundary from '../ErrorBoundary/';
import Search from '../Search/';
import ItemsList from '../ItemsList/';
import MovieDetails from '../MovieDetails/';

class App extends Component {
    constructor(props) {
        super(props);
        this.handleMovieClick = (id) => {
            let clickedMovie = this.props.moviesList.find((movieItem) => id == movieItem.id);
            this.props.onShowMovie(clickedMovie);
        };
    }

    render() {
        return (
            <div>
                <em>details: {this.props.info}</em>
                <ErrorBoundary>
                    { !this.props.chosenMovie ?
                        <Search/> :
                        <MovieDetails movie={this.props.chosenMovie}/>
                    }
                </ErrorBoundary>

                <ErrorBoundary>
                    <ItemsList
                        handleMovieClick={this.handleMovieClick}/>
                </ErrorBoundary>
            </div>
        );
    }
}

export default connect(
    (state) => {
        return {
            moviesList: state.moviesList,
            chosenMovie: state.chosenMovie,
        };
    },
    (dispatch) => {
        return {
            onShowMovie: (clickedMovie) => dispatch(showMovie(clickedMovie)),
        }
    },
)(App);