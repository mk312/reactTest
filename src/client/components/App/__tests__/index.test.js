import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import { reducer } from './../../../reducers/reducer';
import ReduxThunk from 'redux-thunk';
import {Simulate} from 'react-dom/test-utils';
import {render, fireEvent} from '@testing-library/react'
import { moviesList } from './../../mockMoviesList';
import App from '../index';

let container = null;
let store  = createStore(reducer, applyMiddleware(ReduxThunk));
describe('App', () => {
    beforeAll(() => {
        const mockSuccessResponse = {data: moviesList.data };
        global.fetch = jest.fn(() =>
            Promise.resolve({
                json: () => Promise.resolve(mockSuccessResponse),
            })
        );
    });
    beforeEach(() => {
        store = createStore(reducer, applyMiddleware(ReduxThunk));
        container = render(
            <Provider store={store}>
                <App info='React Test App, PartI'/>
            </Provider>
        ).container;
    });

    it('should show movieDetails when movieItem is clicked()', () => {
        const button = document.querySelector('.js-movie-item a');
        fireEvent.click(
            button, new Event('click')
        );

        expect(document.querySelectorAll('.js-movie-details').length).toBe(1);
        // re-check with snapshot
        expect(container).toMatchSnapshot();
    });

    it('should show itemsList result when new search value is submitted', () => {
        const searchInput = document.querySelector('.js-search-input');
        const searchForm = document.querySelector('.js-search-form');
        const searchValues = [
            'Quentin Tarantino',
            'noMatchesRequest'
        ]
        searchValues.forEach((searcString)=>{
            fireEvent.change(searchInput, { target: { value: searcString}});
            fireEvent.submit(
                searchForm
            );
            const moviesNumber = document.querySelectorAll('.js-movie-item').length;

            expect(moviesNumber).toBe(moviesList.data.length);
            // re-check with snapshot
            expect(container).toMatchSnapshot();
        })
    });
});