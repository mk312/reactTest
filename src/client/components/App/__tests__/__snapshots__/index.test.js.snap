// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should show itemsList result when new search value is submitted 1`] = `
<div>
  <div>
    <em>
      details: 
      React Test App, PartI
    </em>
    <form
      class="js-search-form wrapper"
    >
      <div
        class="row"
      >
        <div
          class="title"
        >
          FIND YOUR MOVIE
        </div>
        <input
          class="js-search-input input"
          type="text"
          value="Quentin Tarantino"
        />
        <input
          class="submit"
          type="submit"
          value="SEARCH"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="filter"
        >
          SEARCH BY 
          TITLE
        </div>
        <input
          checked=""
          class="js-toggle-radio-button"
          id="filterValue1"
          name="contact"
          type="radio"
          value="TITLE"
        />
        <label
          for="filterValue1"
        >
          TITLE
        </label>
        <input
          class="js-toggle-radio-button"
          id="filterValue2"
          name="contact"
          type="radio"
          value="GENRE"
        />
        <label
          for="filterValue2"
        >
          GENRE
        </label>
      </div>
    </form>
    <div
      class="result"
    >
      <div
        class="number"
      >
        2
         movies found
      </div>
      <div
        class="sort"
      >
        <div
          class="sortTitle"
        >
          SORT BY
        </div>
        <input
          checked=""
          class="js-toggle-radio-button"
          id="sv1"
          name="contact"
          type="radio"
          value="DATE"
        />
        <label
          for="sv1"
        >
          DATE
        </label>
        <input
          class="js-toggle-radio-button"
          id="sv2"
          name="contact"
          type="radio"
          value="RATING"
        />
        <label
          for="sv2"
        >
          RATING
        </label>
      </div>
    </div>
    <div
      class="list"
    >
      <div
        class="js-movie-item movie"
      >
        <a
          href="#"
        >
          <img
            class="poster"
            src="media/51fY8NCyXpL._SX342_.jpg"
          />
        </a>
        <div
          class="title"
        >
          Kill Bill-1
        </div>
        <div
          class="release"
        >
          2000
        </div>
        <div
          class="genre"
        >
          Drama
        </div>
      </div>
      <div
        class="js-movie-item movie"
      >
        <a
          href="#"
        >
          <img
            class="poster"
            src="media/51fY8NCyXpL._SX342_.jpg"
          />
        </a>
        <div
          class="title"
        >
          Kill Bill-2
        </div>
        <div
          class="release"
        >
          2002
        </div>
        <div
          class="genre"
        >
          Drama, Thriller
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App should show itemsList result when new search value is submitted 2`] = `
<div>
  <div>
    <em>
      details: 
      React Test App, PartI
    </em>
    <form
      class="js-search-form wrapper"
    >
      <div
        class="row"
      >
        <div
          class="title"
        >
          FIND YOUR MOVIE
        </div>
        <input
          class="js-search-input input"
          type="text"
          value="noMatchesRequest"
        />
        <input
          class="submit"
          type="submit"
          value="SEARCH"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="filter"
        >
          SEARCH BY 
          TITLE
        </div>
        <input
          checked=""
          class="js-toggle-radio-button"
          id="filterValue1"
          name="contact"
          type="radio"
          value="TITLE"
        />
        <label
          for="filterValue1"
        >
          TITLE
        </label>
        <input
          class="js-toggle-radio-button"
          id="filterValue2"
          name="contact"
          type="radio"
          value="GENRE"
        />
        <label
          for="filterValue2"
        >
          GENRE
        </label>
      </div>
    </form>
    <div
      class="result"
    >
      <div
        class="number"
      >
        2
         movies found
      </div>
      <div
        class="sort"
      >
        <div
          class="sortTitle"
        >
          SORT BY
        </div>
        <input
          checked=""
          class="js-toggle-radio-button"
          id="sv1"
          name="contact"
          type="radio"
          value="DATE"
        />
        <label
          for="sv1"
        >
          DATE
        </label>
        <input
          class="js-toggle-radio-button"
          id="sv2"
          name="contact"
          type="radio"
          value="RATING"
        />
        <label
          for="sv2"
        >
          RATING
        </label>
      </div>
    </div>
    <div
      class="list"
    >
      <div
        class="js-movie-item movie"
      >
        <a
          href="#"
        >
          <img
            class="poster"
            src="media/51fY8NCyXpL._SX342_.jpg"
          />
        </a>
        <div
          class="title"
        >
          Kill Bill-1
        </div>
        <div
          class="release"
        >
          2000
        </div>
        <div
          class="genre"
        >
          Drama
        </div>
      </div>
      <div
        class="js-movie-item movie"
      >
        <a
          href="#"
        >
          <img
            class="poster"
            src="media/51fY8NCyXpL._SX342_.jpg"
          />
        </a>
        <div
          class="title"
        >
          Kill Bill-2
        </div>
        <div
          class="release"
        >
          2002
        </div>
        <div
          class="genre"
        >
          Drama, Thriller
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App should show movieDetails when movieItem is clicked() 1`] = `
<div>
  <div>
    <em>
      details: 
      React Test App, PartI
    </em>
    <div
      class="js-movie-details wrapper"
    >
      <img
        class="poster"
        src="media/51fY8NCyXpL._SX342_.jpg"
      />
      <div
        class="title"
      >
        Kill Bill-1
                
        <div
          class="rating"
        >
          5
        </div>
      </div>
      <div>
        Oscar winning
      </div>
      <div
        class="release"
      >
        2000
         
      </div>
      <div
        class="runtime"
      >
        154
         min
      </div>
      <div
        class="overview"
      >
         American martial arts film written and directed by Quentin Tarantino. It stars Uma Thurman as the Bride, who swears revenge on a team of assassins (Lucy Liu, Michael Madsen, Daryl Hannah, and Vivica A. Fox) and their leader, Bill (David Carradine), after they try to kill her and her unborn child. Her journey takes her to Tokyo, where she battles the yakuza.
         min
      </div>
    </div>
    <div
      class="result"
    >
      <div
        class="number"
      >
        2
         movies found
      </div>
      <div
        class="sort"
      >
        <div
          class="sortTitle"
        >
          SORT BY
        </div>
        <input
          checked=""
          class="js-toggle-radio-button"
          id="sv1"
          name="contact"
          type="radio"
          value="DATE"
        />
        <label
          for="sv1"
        >
          DATE
        </label>
        <input
          class="js-toggle-radio-button"
          id="sv2"
          name="contact"
          type="radio"
          value="RATING"
        />
        <label
          for="sv2"
        >
          RATING
        </label>
      </div>
    </div>
    <div
      class="list"
    >
      <div
        class="js-movie-item movie"
      >
        <a
          href="#"
        >
          <img
            class="poster"
            src="media/51fY8NCyXpL._SX342_.jpg"
          />
        </a>
        <div
          class="title"
        >
          Kill Bill-1
        </div>
        <div
          class="release"
        >
          2000
        </div>
        <div
          class="genre"
        >
          Drama
        </div>
      </div>
      <div
        class="js-movie-item movie"
      >
        <a
          href="#"
        >
          <img
            class="poster"
            src="media/51fY8NCyXpL._SX342_.jpg"
          />
        </a>
        <div
          class="title"
        >
          Kill Bill-2
        </div>
        <div
          class="release"
        >
          2002
        </div>
        <div
          class="genre"
        >
          Drama, Thriller
        </div>
      </div>
    </div>
  </div>
</div>
`;
